manifests/repository.pp

```diff
@@ -17,15 +17,22 @@
 #   The major bareos release version which should be used
 # @param password
 #   The major bareos release version which should be used
+# @param https
+#   Wether or not https should be used in repo URL
 #
 class bareos::repository (
   String              $release             = 'latest',
   Optional[String[1]] $gpg_key_fingerprint = undef,
   Boolean             $subscription        = false,
   Optional[String]    $username            = undef,
   Optional[String]    $password            = undef,
+  Boolean             $https               = true,
 ) {
-  $scheme = 'http://'
+  if $https {
```

Comment:
While your code works, this is more clear to the reader.

```puppet
if $https == true {
```

Reply:
@ghoneycutt that's entirely redundant, nonstandard (for any language, not just puppet), and not supported by the Style Guide or other documentation. IMO it's an antipattern, redundant code should be avoided. More code means more potential for problems.

Reply:
Having spent a lot of time teaching people with no coding skills how to model their infrastructure with Puppet, this helps them. Anyhow, it was just an observation and not meant as something to be done to pass the review.